### DISCUSSION TITLE
Content Indexers

### DISCUSSION THREAD

**Discussion by abueide** [n0-computer/iroh#1424] [Ideas]
Hi, so I see that ipfs is kind of banking on IPNI to reduce their content resolution times. Does iroh have any sort of content indexer planned or are you guys all in on building a super fast scalable dht? I think indexers are a really interesting supplement to the DHT especially in scenarios of federated communities kind of like mastodon or lemmy subbreddits where a lot of the users in that community will be accessing & sharing similar content and there is sort of community hub that can do indexing and help peers find each other with low latency for media / browsing purposes.

**Comment by b5** [comment]
Yeah, we've put off plans to build a DHT for now, and are planning to have network anchors function as indexers. 

Coming from IPFS space, you can think of anchor nodes as:
* pinning + content indexing
* one per app
* arranged in a hierarchy (anchors can have "parents" they defer up to when content isn't found)

so yes, short answer is we're super into the idea of an indexer, to the point that we're building it first, and deferring work on a DHT until we hit scales where it makes sense
